### PR TITLE
Migrate  DeckPickerTest to kotlin

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description

Migrate DeckPickerTest.java to kotlin following the migration guide.

## How Has This Been Tested?

Ran the tests.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt [hunk 1]
```diff
@@ -1,583 +1,650 @@
...
+            MatcherAssert.assertThat(
+                "A specific dialog for a conflict should be shown",
+                d.databaseErrorDialog,
+                Matchers.`is`(DatabaseErrorDialog.DIALOG_DB_LOCKED)
+            )
         } finally {
-            BackendEmulatingOpenConflict.disable();
-            InitialActivityWithConflictTest.setupForDefault();
+            disable()
+            setupForDefault()
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt:296]: These should be moved back in

File: AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt [hunk 2]
```diff
@@ -1,583 +1,650 @@
...
 
     @Test
-    public void deckPickerOpensWithHelpMakeAnkiDroidBetterDialog() {
+    fun deckPickerOpensWithHelpMakeAnkiDroidBetterDialog() {
         // Refactor: It would be much better to use a spy - see if we can get this into Robolectric
         try {
-            InitialActivityWithConflictTest.grantWritePermissions();
-            BackupManagerTestUtilities.setupSpaceForBackup(getTargetContext());
+            grantWritePermissions()
+            setupSpaceForBackup(targetContext)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt:305]: ditto

**Review by david-allison — CHANGES_REQUESTED** [review]
Looks great! A couple of imports were removed

File: AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt [hunk 3]
```diff
@@ -1,583 +1,641 @@
...
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import org.robolectric.ParameterizedRobolectricTestRunner
+import org.robolectric.Robolectric
+import org.robolectric.RuntimeEnvironment
+import java.io.File
+import java.util.*
```
Comment (divyansh-dxn) [inline AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt:32]: I think wildcard import can be avoided for util.

File: AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt [hunk 4]
```diff
@@ -1,583 +1,641 @@
...
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+@RunWith(ParameterizedRobolectricTestRunner::class)
+@KotlinCleanup("fix IDE lint issues")
+@KotlinCleanup("replace `when` usages")
```
Comment (divyansh-dxn) [inline AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt:38]: replace `when` and `is` usages

**Review by david-allison — APPROVED** [review]

File: AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt (hunk 3 above)
Comment (lukstbit) [reply AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt:32]: I'm going to leave it as it is until #11852  is done.

File: AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt (hunk 4 above)
Comment (lukstbit) [reply AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.kt:38]: I missed that.

**Review by mikehardy — APPROVED** [review]
Thanks!

**Pull request merged**